src/mui/input/EmbeddedArrayInput.js

```diff
@@ -0,0 +1,161 @@
+import FlatButton from 'material-ui/FlatButton';
+import React, { Component } from 'react';
+import { FieldArray, Field } from 'redux-form';
+import PropTypes from 'prop-types';
+
+import ContentCreateIcon from 'material-ui/svg-icons/content/create';
+import ActionDeleteIcon from 'material-ui/svg-icons/action/delete';
+
+/**
+ * An Input component for generating/editing an embedded array
+ *
+ *
+ * Use it with any set of input componentents as children, like `<TextInput>`,
+ * `<SelectInput>`, `<RadioButtonGroupInput>` ... etc.
+ *
+ * @example
+ * export const CommentEdit = (props) => (
+ *     <Edit {...props}>
+ *         <SimpleForm>
+ *              <EmbeddedArrayInput source="links">
+ *                  <SimpleList>
+ *                       <TextField source="url" />
+ *                       <TextField source="context"/>
+ *                   </SimpleList>
+ *               </EmbeddedArrayInput>
+ *         </SimpleForm>
+ *     </Edit>
+ * );
+ */
+export class EmbeddedArrayInput extends Component {
+    constructor(props) {
+        super(props);
+        // nothing to do for now
+    }
+
+    renderList = ({ fields }) => {
```

Comment:
This is not how `FieldArray` is supposed to work. The signature should be

`renderList = (items) =>`

And then you must iterate over items, not fields. Otherwise, it doesn't work (and in your code, when I click on the add button, it duplicates the current item).